### DISCUSSION TITLE
Question about Multiple Relationships

### DISCUSSION THREAD

**Discussion by epulla** [pocketbase/pocketbase#5461] [Q&A] [answered]
hey there 👋 first of all, thanks a lot for all the hard work put in this project, it's been amazing!!

## Context
I am creating a real-time chat app using Pocketbase and I have these relationships (simplified):
```mermaid
erDiagram
chat_sessions {
  id string
}
chat_messages {
  id string
  message string
  more fields
}
chat_sessions ||--|{ chat_messages : ""
```
where a __chat_session__ has multiple __chat_messages__.

## Question
In Pocketbase (I'm using v0.22.19), I know that the Multiple Relationships are handled by adding a JSON array as a table field in the db, which is fine for me, but I am wondering if there would be a limit of bytes (or any kind of limits/caveats) for the JSON table field in the case where there are hundreds (or in the worst case, thousands) of __chat_messages__ for a single __chat_session__?

Or if it would be better to normalize the tables/collections for this case? (e.g. add a single relationship field in each __chat_message__ to have a reference to a single __chat_session__)

Thank you!

**Comment by ganigeorgiev** [comment, marked as answer]
If you are planning to have thousands of messages in single sessions then it would be better to restructure your collections and either add a `relation` field from `chat_messages` -> `chat_sessions` as you've mentioned OR in case a single message needs to be associated to more than one session - move the relationship to a separate junction collection that will have 2 single `relation` fields pointing to the 2 collections. Adding an index for the relation field(s) could help additionally with the performance.

This will be documented better with the next release but in general the multiple `relation` field was usually intended to store small/finite number of related items like labels, categories, tags, groups, roles, etc. without the need of junction collection.

The current "unlimited" `MaxSelect` field option was a bad decision and it is already reverted as part of the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) to its old state from earlier versions to be always required for more consistent behavior with the other multivalued fields.

**Reply by epulla** [reply]
Thanks for the quick response @ganigeorgiev! you're awesome sir! ❤️
